### PR TITLE
Add DbCollection Add Signup Model in User.js

### PR DESCRIPTION
**Pull request by Neeraj-Bora**

### CODE REVIEW

File: src/db/DbCollection.js
Comment (hereisadi) [inline src/db/DbCollection.js:1]: There is no need to create this DbCollection.js file. The schema which you have declared should be in models folder.

File: src/models/User.js
Comment (hereisadi) [inline src/models/User.js:1]: You have to define the schema and models here in this file.
Define the signup function in Controllers folder.

File: src/models/User.js [hunk 1]
```diff
@@ -1 +1,29 @@
...
+        email: req.body.email
+    }
+    await collection.insertMany([data]);
+
+    res.json({
+        message:"posted successfully"
+    });
+});
+
+app.listen(3000,()=>{
```
Comment (hereisadi) [inline src/models/User.js:26]: app.listen is done in index.js and need not to be done in every individual file.

File: src/models/User.js
Comment (hereisadi) [inline src/models/User.js:1]: Use try catch block to properly handle errors with status code.
Moreover, throw an error "user already exists" if user has already created an account, use bcrypt to hash the password before storing it in db.

**Review by hereisadi — CHANGES_REQUESTED** [review]

**Comment by Himangshu-101** [comment]
> _ No description provided. _

Write proper descriptions about your commits.  If you see merge conflicts in your PR, then take the necessary actions to remove those conflicts too !

File: src/controllers/UserController.js [hunk 2]
```diff
@@ -17,11 +17,67 @@
...
+  });
+
+  const SignUp= new mongoose.model("MONGODBSECRET",SignUpSchema);
+
+  app.post("/signup",async(req,res)=>{
+      try{
+
+          const userExist= await SignUp.findOne({email: req.email});
+          if(userExist){
+              return res.status(422).json({error:"User already exists"});
```
Comment (hereisadi) [inline src/controllers/UserController.js:44]: Preferred status code should be 400 in this case. Please go through https://developer.mozilla.org/en-US/docs/Web/HTTP/Status and carry out changes if required.

File: src/controllers/UserController.js [hunk 3]
```diff
@@ -17,11 +17,67 @@
...
+
+        this.password= await bcrypt.hash(this.password, 10);
+        this.confirmpassword= await bcrypt.hash(this.confirmpassword,10);
+    }
+    next();
+  });
+
+  const SignUp= new mongoose.model("MONGODBSECRET",SignUpSchema);
+
+  app.post("/signup",async(req,res)=>{
```
Comment (hereisadi) [inline src/controllers/UserController.js:39]: you have already created `router.post("/signup", )` in UserRoutes.js. Here, you need only to define the signup function. Correct it like below:
`const signup = (req,res) => { }`

**Review by hereisadi — CHANGES_REQUESTED** [review]
image
Fix these errors, and we will be okay to proceed further with this PR

File: src/controllers/UserController.js [hunk 4]
```diff
@@ -17,11 +17,63 @@
...
+          }
+
+        
+      }catch(error){
+          res.status(400).send(error);
+      }
+  
+};
+
+module.exports = {
```
Comment (Himangshu-101) [inline src/controllers/UserController.js:72]: write module exports as one time only... below in the file it is already used.
So export signUp there only (home is already exported) and remove this one

**Review by Himangshu-101 — CHANGES_REQUESTED** [review]

File: src/routes/UserRoutes.js [hunk 5]
```diff
@@ -3,6 +3,7 @@ const router = express.Router();
 const userController = require('../controllers/UserController');
 
 router.get('/', userController.home);
+router.post("/signup", userController.UserRoutes.signUp)
```
Comment (hereisadi) [inline src/routes/UserRoutes.js:6]: it should be `router.post("/signup", userController.signUp)`

**Review by hereisadi — CHANGES_REQUESTED** [review]

File: src/models/User.js [hunk 6]
```diff
@@ -1 +1,41 @@
 // signup model goes here
+
+const express= require('express');
+const app= express();
```
Comment (hereisadi) [inline src/models/User.js:4]: no need to importing express in model file

**Review by hereisadi — APPROVED** [review]

**Review by Himangshu-101 — APPROVED** [review]

**Pull request closed without merging**